### ISSUE TITLE
Include Stash element in Javascript if condition

### ISSUE THREAD

**Issue by cprudhoe** [FMCorz/moodle-block_stash#27]
Is there a way to use the Stash link [stashdrop secret = "x0GUV" text = "Pick up!" image] in a JavaScript if condition?

I want to show a different stash image based on the user's score.

if (score == 100) {
   [stashdrop secret = "x0GUV" text = "Pick up!" image]
}

I can write the JavaScript code to get the user's score and test the condition, but I can't find a way to call the stashdrop from inside the JavaScript code.

Thanks for your help,
Clarence Prudhoe
cprudhoe@musapp.com

**Comment by FMCorz** [comment]
Hi @cprudhoe,

This will be a bit challenging, but it might be possible. The shortcode is converted to HTML and JavaScript instructions. If your JavaScript instruction is passed through Moodle filters (which is likely not), it will be converted to something like this:

```js
if (score == 100) {
  <div>...</div>
}
```

This is obviously not valid JavaScript.

You may have more success by wrapping the shortcode in a div, and changing that divs visibility as per the score in your JavaScript scripts.

```html
<div class="hidden" id="a">[startdrop 1]</div>
<div class="hidden" id="b">[startdrop 2]</div>
```

```js
if (score == 100) {
  document.getElementById('a').classList.remove('hidden');
}
````

I hope this helps. Meanwhile please note that the plugin is currently maintained here: https://github.com/abgreeve/moodle-block_stash

**Comment by cprudhoe** [comment]
Hi Frederic,
Thank you so much for this code idea.  This works great!
I have developed a very large project to help users develop musical 
performance skills.  MusAPP.club listens to your live performance via 
microphone and shows your pitch in relation to printed music notation.

There are over 1450 lessons that judge your performance and scores you 
based on how long it takes you to get on pitch and if you sing or play 
the rhythms as notated.

I am pleased that I can use Moodle-block_Stash to encourage user 
participation.
I needed to be able to read the performance score in a Moodle lesson and 
point to a relivant Stash image.

Thanks again for your prompt response and greate programming idea.
Clarence Prudhoe
***@***.***
www.musapp.club
